src/packages/api-client/index.ts

```diff
@@ -8,9 +8,9 @@ declare module 'axios' {
 
 type ApiClientParams = {
   axiosConfig: AxiosRequestConfig;
-  selectToken: () => string | undefined;
-  onApiCallStart: (config: AxiosRequestConfig) => void;
-  onApiCallFinish: (config: AxiosRequestConfig) => void;
+  selectToken: () => string | null;
+  onApiCallStart: (apiCallId: string) => void;
```

Comment:
Now it's not possible to send other values for possible future features. Is that really what we want?

Reply:
ake? si len vymyslal 🤣

Reply:
a druha vec, spinner je separatny package od api-client packagu, nemali by o sebe vobec vediet. teraz si ich tight couplol. to bola hlavna pointa, preco som posielal config, nie konkretny value.

Reply:
nevedia o sebe, neni coupling, config je zbytocne posielat, naco?